KubernetesClientException if Wildcard (*) in NO_PROXY environment variable

### Describe the bug

If the NO_PROXY environment variable contains wildcards, a KubernetesClientException is thrown when attempting to create a KubernetesClient. 

`KubernetesClient kubernetesClient = new KubernetesClientBuilder().build();`

If the NO_PROXY environment variable has wildcards, for example ***.domain.com**, then client creation is failing with the below exception.

`io.fabric8.kubernetes.client.KubernetesClientException: Invalid proxy server configuration`
`java.net.MalformedURLException: NO_PROXY URL contains invalid entry: '*.domain.com'`

### Fabric8 Kubernetes Client version

6.1.1

### Steps to reproduce

1. Set wildcards in the NO_PROXY environment variable
`NO_PROXY: localhost,127.0.0.1,*.google.com, *.github.com`
3. Try to create a Kubernetes client

### Expected behavior

We should be able to create Kubernetes clients even when the NO_PROXY environment variable contains wildcard expressions.

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

other (please specify in additional context)

### Environment

Windows

### Fabric8 Kubernetes Client Logs

```shell
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Invalid proxy server configuration
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.applyCommonConfiguration(HttpClientUtils.java:184)
	at io.fabric8.kubernetes.client.okhttp.OkHttpClientFactory.createHttpClient(OkHttpClientFactory.java:96)
	at io.fabric8.kubernetes.client.okhttp.OkHttpClientFactory.createHttpClient(OkHttpClientFactory.java:32)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:149)
	at io.fabric8.kubernetes.client.DefaultKubernetesClient.<init>(DefaultKubernetesClient.java:163)
	... 78 more
Caused by: java.net.MalformedURLException: NO_PROXY URL contains invalid entry: '*.domain.com'
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.isHostMatchedByNoProxy(HttpClientUtils.java:206)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.getProxyUrl(HttpClientUtils.java:64)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.applyCommonConfiguration(HttpClientUtils.java:173)
	... 82 more
```


### Additional context

`
[opc@fabric-demo-bastion-2 ~]$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:44:59Z", GoVersion:"go1.19", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"c999fea1e6dbf111d64000dfaaff45edaa4c34d2", GitTreeState:"clean", BuildDate:"2022-01-11T12:16:14Z", GoVersion:"go1.16.12 BoringCrypto", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.25) and server (1.22) exceeds the supported minor version skew of +/-1
`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -186,6 +186,7 @@ fix #4373: NO_PROXY should allow URIs with hyphens ("circleci-internal-outer-bui
   - `io.fabric8.chaosmesh.v1alpha1.AwsChaos` => `io.fabric8.chaosmesh.v1alpha1.AWSChaos`
   - `io.fabric8.chaosmesh.v1alpha1.IoChaos` => `io.fabric8.chaosmesh.v1alpha1.IOChaos`
   - `io.fabric8.chaosmesh.v1alpha1.PodIoChaos` => `io.fabric8.chaosmesh.v1alpha1.PodIOChaos`
+* Fix #4247: Proxy matching no longer supports having wildcard characters in `NO_PROXY`. The behavior has been changed to match [GNU WGet Spec](https://www.gnu.org/software/wget/manual/html_node/Proxies.html)
 
 ### 5.12.3 (2022-07-27)
 

--- a/doc/FAQ.md
+++ b/doc/FAQ.md
@@ -102,3 +102,20 @@ Once added KubernetesClient declarative services would be exposed automatically 
   @BeanInject
   private KubernetesClient kubernetesClient
 ```
+
+### Why am I getting Exception when using `*` in NO_PROXY ?
+
+Starting Fabric8 Kubernetes Client v6.1.0, we've change NO_PROXY matching as simple as possible and not support any meta characters. It honors the [GNU WGet Spec](https://www.gnu.org/software/wget/manual/html_node/Proxies.html).
+
+So instead of providing NO_PROXY like this:
+
+(Unsupported) :x:
+```java
+NO_PROXY: localhost,127.0.0.1,*.google.com, *.github.com
+```
+we should provide it like this:
+
+(Supported) :heavy_check_mark:
+```java
+NO_PROXY: localhost,127.0.0.1,.google.com,.github.com
+```

